main.py

```diff
@@ -1,19 +1,24 @@
-import http.client
-import json
+from reader import HttpRequestResponse
 import datetime
 from util import SortUtil
 
-print('Start')
+#get current time
 start = datetime.datetime.now()
-conn = http.client.HTTPSConnection('api.spaceflightnewsapi.net')
-headers = {'Content-type': 'application/json'}
-conn.request("GET", "/v3/articles?_limit=30", headers=headers)
-response = conn.getresponse()
-js = json.loads(response.read().decode())
+
+#create instance of a class
+request_response = HttpRequestResponse('api.spaceflightnewsapi.net', '/v3/articles?_limit=30')  
+
+#retrieve data      
+data_list = request_response.establish_connection('api.spaceflightnewsapi.net', '/v3/articles?_limit=30', start)
```

Comment:
Passing 'api.spaceflightnewsapi.net' and '/v3/articles?_limit=30' in establish_connection is redundant because you have passed these values in the constructor.
You need to choose either to pass the values on the constructor or establish_connection function.

Reply:
link and request parameters have been removed from establish_connection function. The values are passed to the constructor